File: server/test/factories/person.factory.ts
import { PersonLike, PersonRow } from 'test/factories/types.js';
import { newDate, newUuid, newUuidV7 } from 'test/small.factory.js';

export class PersonFactory {
  private constructor(private readonly value: PersonRow) {}

  static create(dto: PersonLike = {}) {
    return PersonFactory.from(dto).build();
  }

  static from(dto: PersonLike = {}) {
    return new PersonFactory({
      birthDate: null,
      color: null,
      createdAt: newDate(),
      faceAssetId: null,
      personGroupId: newUuid(),
      isFavorite: false,
      isHidden: false,
      name: 'person',
      otherPeople: [],
      sharedBy: [],
      sharedWith: [],
      ownerId: newUuid(),
      thumbnailPath: '/data/thumbs/person-thumbnail.jpg',
      updatedAt: newDate(),
      updateId: newUuidV7(),
      ...dto,
    });
  }

  build() {
    return { ...this.value };
  }
}

File: mobile/test/unit/factories/person_factory.dart
import 'package:immich_mobile/domain/models/person.model.dart';

import '../../utils.dart';

class PersonFactory {
  static DriftPerson create({String? id, String? name}) {
    id ??= TestUtils.uuid();
    name ??= 'person_$id';
    return DriftPerson(
      id: id,
      name: name,
      createdAt: TestUtils.date(),
      updatedAt: TestUtils.date(),
      isFavorite: TestUtils.randDouble() > 0.5,
      isHidden: TestUtils.randDouble() > 0.5,
      birthDate: TestUtils.date(),
    );
  }
}
